vortex-expr/src/literal.rs

```diff
@@ -37,6 +37,51 @@ impl Display for Literal {
     }
 }
 
+#[cfg(feature = "proto")]
+pub(crate) mod proto {
+    use kind::Kind;
+    use vortex_error::{VortexResult, vortex_bail, vortex_err};
+    use vortex_proto::expr;
+    use vortex_proto::expr::kind;
+    use vortex_scalar::Scalar;
+
+    use crate::{ExprDeserialize, ExprRef, ExprSerializable, Id, Literal};
+
+    pub(crate) struct LiteralSerde;
+
+    impl Id for LiteralSerde {
```

Comment:
Could consider representing the ids as enum. Then we could also index into the `EXPRESSIONS` array directly without having to build a hashmap.

Reply:
I want to support extensions in the future and an enum will not allow that